src/software/backend/simulation/physics_field.cpp

```diff
@@ -0,0 +1,110 @@
+#include "software/backend/simulation/physics_field.h"
+
+#include "software/backend/simulation/box2d_util.h"
+
+PhysicsField::PhysicsField(std::shared_ptr<b2World> world, const Field &field)
+    : field(field)
+{
+    setupFieldBoundary(world, field);
+    setupEnemyGoal(world, field);
+    setupFriendlyGoal(world, field);
+}
+
+void PhysicsField::setupFieldBoundary(std::shared_ptr<b2World> world, const Field &field)
+{
+    field_boundary_body_def.type = b2_staticBody;
+    // Note that the body shape is defined relative to the body position. Setting the
+    // body position to (0, 0) makes it easy to add the shape using the standard field
+    // interface, since field coordinates are also relative to (0, 0)
+    // https://www.iforce2d.net/b2dtut/fixtures
+    field_boundary_body_def.position.Set(0, 0);
+    field_boundary_body = world->CreateBody(&field_boundary_body_def);
+
+    const unsigned int shape_loop_size = 4;
+    b2Vec2 vs[shape_loop_size]         = {
```

Comment:
what does `vs` mean? Could we give it a more descriptive name? Same goes for `shape_loop_size`? Perhaps `field_boundary_vertices` and `num_field_boundary_vertices` (assuming I'm reading this correctly)?

Reply:
Whoops, you're right. I wasn't paying attention and probably took this from example code. Fixed in the way you suggested.